### ISSUE TITLE
About GPU utilization

### ISSUE THREAD

**Issue by Hzj199** [kohya-ss/sd-scripts#965]
In multi-machine, multi-GPU training, the InfiniBand (IB) network shows no traffic, suggesting that avg_loss = accelerator.gather(loss).mean() might not be used.

**Comment by Hzj199** [comment]
Why is it necessary to use transform_models_if_DDP(models) after accelerator.prepare?

**Comment by Isotr0py** [comment]
`transform_models_if_DDP(models)` is a wrong writing made by mistake and we haven't noticed it before.  This will be removed in the new PR, because it breaks the gradients sync.

I think this may also break the multi-machine, multi-GPU communication. Maybe the new PR can fix this.

I only have two GPUs on one machine and can't test in multi-machine training.  So could you check if the update works on the multi-machine, multi-GPU training?
